File: 3732-calculate-score-after-performing-instructions/3732-calculate-score-after-performing-instructions.go
func calculateScore(instructions []string, values []int) int64 {
    n := len(values)
    vis := make([]bool, n)
    var sum int64
    sum = 0
    for i := 0; i < n && i >= 0 && !vis[i]; {
        vis[i] = true
        if instructions[i] == "add" {
            sum += int64(values[i])
            i++
        } else {
            i += values[i]
        }
    }
    return sum
}

File: 3732-calculate-score-after-performing-instructions/3732-calculate-score-after-performing-instructions.js
/**
 * @param {string[]} instructions
 * @param {number[]} values
 * @return {number}
 */
var calculateScore = function(instructions, values) {
    let n = values.length;
    let sum = 0;
    let s = new Set()
    for (let i = 0; i < n && i >= 0 && !s.has(i); ) {
        s.add(i);
        if (instructions[i] == "add") {
            sum += values[i];
            i++;
        }
        else
            i += values[i];
    }
    return sum;
};